src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -0,0 +1,2201 @@
+				.collect(Collectors.toList());
+			if (!invRowWithoutRealQty.isEmpty()) {
+				MessageDialog.error(null, "angal.inventory.allinventoryrowswithnewlotshouldhaverealqtygreatterthanzero.msg");
+				return;
+			}
+			String errorMessage = this.checkParamsValues(chargeCode, dischargeCode, supplierId, wardCode);
+			if (errorMessage != null) {
+				MessageDialog.error(null, errorMessage);
+				return;
+			}
+			if (checkParametersChanges(wardCode, chargeCode, dischargeCode, supplierId, lastReference, lastInventoryDate)) {
+				MessageDialog.error(null, "angal.inventory.pleasesaveinventorybeforevalidateit.msg");
+				return;
+			}
+			// validate inventory
+			String status = InventoryStatus.validated.toString();
+			String option = askAllMedicalsOrList("angal.inventory.doyouwanttocheckforallmedicalsinthestockoronlytheonesinthelist.msg");
+			if (option == null) {
+				return;
+			}
+			allMedicals = option.equals(allMedicalsOrList[0]); // Yes (All medicals)
+			allMedicalsChosen = true;
+			try {
+				medicalInventoryManager.validateMedicalInventoryRow(inventory, inventoryRowSearchList, allMedicals);
+				inventory.setStatus(status);
+				inventory = medicalInventoryManager.updateMedicalInventory(inventory, true);
+				MessageDialog.info(null, "angal.inventory.validate.success.msg");
+				statusLabel.setText(status.toUpperCase());
+				statusLabel.setForeground(Color.BLUE);
+				confirmButton.setEnabled(true);
+				fireInventoryUpdated();
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+				int answer = MessageDialog.yesNo(null, "angal.inventory.doyouwanttoactualizetheinventory.msg");
+				if (answer == JOptionPane.YES_OPTION) {
+					try {
+						inventory.setStatus(status);
+						inventory = medicalInventoryManager.actualizeMedicalInventoryRow(inventory, allMedicals);
+						dateInventory = inventory.getInventoryDate();
+						statusLabel.setText(status.toUpperCase());
+						statusLabel.setForeground(Color.BLUE);
+						confirmButton.setEnabled(true);
+						jTableInventoryRow.setModel(new InventoryRowModel());
+						fireInventoryUpdated();
+					} catch (OHServiceException e1) {
+						OHServiceExceptionUtil.showMessages(e1);
+					}
+				} else {
+					try {
+						inventory.setStatus(InventoryStatus.draft.toString());
+						statusLabel.setText(InventoryStatus.draft.toString().toUpperCase());
+						statusLabel.setForeground(Color.GRAY);
+						inventory = medicalInventoryManager.updateMedicalInventory(inventory, true);
+						fireInventoryUpdated();
+					} catch (OHServiceException ex) {
+						OHServiceExceptionUtil.showMessages(ex);
+					}
+				}
+			}
+		});
+		return validateButton;
+	}
+
+	private String askAllMedicalsOrList(String question) {
+		String option = (String) MessageDialog.inputDialog(null, null, allMedicalsOrList, null, question);
+		return option;
```

Comment:
there is no need to assign to a local variable; just immediately return the result of the expression

Reply:
Done, thanks!